src/util/mod.rs

```diff
@@ -547,14 +548,20 @@ impl<T: ?Sized> AsAddress for *mut T {
     }
 }
 
-/// Is `t` aligned to `align_of::<U>()`?
+/// Validates that `t` is aligned to `align_of::<U>()`.
 #[inline(always)]
-pub(crate) fn aligned_to<T: AsAddress, U>(t: T) -> bool {
+pub(crate) fn validate_aligned_to<T: AsAddress, U>(t: T) -> Result<(), AlignmentError<(), U>> {
```

Comment:
Take it or leave it:
```suggestion
pub(crate) fn validate_aligned_to<T: AsAddress, U>(t: T) -> Result<(), AlignmentError<T, U>> {
```

Reply:
I tried that, but it ends up being worse because no callers actually pass a byte slice directly - they pass things like `bytes.deref()` - and so all of them would end up needing to do `.with_src` anyway.

Also, this would require changing `AsAddress::addr` to take a `&self` receiver - otherwise, calling `t.addr()` consumes the `t` and so it can't be returned.